DEVELOPING.md

```diff
@@ -0,0 +1,95 @@
+# Developing Airflow Rocks
+
+This document describes how to build, test, and validate the Apache Airflow Rocks packaged in this repository.
+
+---
+## Installing dependencies
+
+The following are dependencies to be able to develop and test changes to the Temporal rocks:
+
+- a K8s distribution (ideally k8s by Canonical)
+- rockcraft
+- yq
+- kubectl
+- just
+- docker
+- goss
+- kgoss
+
+There are convenient snaps for all of the above dependencies besides goss and kgoss. The recommended way to install these until a goss snap is released would be:
+```bash
+goss_base_url="https://github.com/goss-org/goss/releases/latest/download"
+curl -L ${goss_base_url}/goss-linux-amd64 -o /usr/local/bin/goss
+chmod +rx /usr/local/bin/goss
+curl -L ${goss_base_url}/kgoss -o /usr/local/bin/kgoss
+chmod +rx /usr/local/bin/kgoss
+```
+
+
+## Repository Structure
+
+
+- **3.1.0/**
+  - `rockcraft.yaml` — Rockcraft build definition for Airflow 3.1.0  
+  - `goss.yaml` — Base health checks  
+  - `goss_wait.yaml` — Startup readiness checks  
+  - `environment` — Version-specific environment variables  
+- `justfile` — Task automation for building, cleaning, and testing  
+- `LICENSE` — Project license file  
+- `README.md` — Project README  
+
+- Each version directory (e.g., `3.1.0/`) contains everything needed to build that Airflow release.
+- Currently, this repository supports **Apache Airflow 3.1.0**, the latest stable upstream version.
```

Comment:
I'd remove the versions from this doc. When we move away from 3.1.0, we'll need to update this, making it a chore to keep up to date.